File: app/src/main/java/com/goel/attendancetracker/NewOrganisationActivity.java
package com.goel.attendancetracker;

import android.content.ContentValues;
import android.content.Intent;
import android.database.sqlite.SQLiteException;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.goel.attendancetracker.database.DatabaseHandler;
import com.goel.attendancetracker.database.OrganisationsDataModel;
import com.goel.attendancetracker.database.Params;

public class NewOrganisationActivity extends AppCompatActivity {

    DatabaseHandler databaseHandler;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_new_organisation);

        databaseHandler = new DatabaseHandler(NewOrganisationActivity.this);
    }


    public void submitNewOrganisation(View view)
    {
        EditText name = findViewById(R.id.edit_new_organisation_name);
        EditText target = findViewById(R.id.edit_new_organisation_target);

        if (name.getText().toString().isEmpty())
        {
            name.setError("Please enter a valid name");
            return;
        }

        if (target.getText().toString().isEmpty())
        {
            target.setError("Please enter a valid number");
            return;
        }

        String organisationName = name.getText().toString();
        int organisationTarget = Integer.parseInt(target.getText().toString());

        if (organisationName.length() > 30)
        {
            name.setError("Name length should be less than or equal to 30");
            return;
        }

        if (organisationTarget < 0 || organisationTarget > 100)
        {
            target.setError("Please enter a number from 0 to 100");
            return;
        }

        OrganisationsDataModel newOrganisation = new OrganisationsDataModel(organisationName, organisationTarget);

        try {
            databaseHandler.createNewOrganisation(newOrganisation);
            insertOrganisation(newOrganisation);
        } catch (SQLiteException e) {
            name.setError("Organisation already exists");
            Toast.makeText(this, "Organisation already exists", Toast.LENGTH_SHORT).show();
        }
    }


    private void insertOrganisation(OrganisationsDataModel newOrganisation) {

        ContentValues values = new ContentValues();
        values.put(Params.NAME, newOrganisation.getName());
        values.put(Params.TARGET, newOrganisation.getTarget());
        values.put(Params.ATTENDANCE, newOrganisation.getAttendance());
        databaseHandler.insertOrganisation(values);

        Toast.makeText(this, "Added " + newOrganisation.getName() + " Successfully", Toast.LENGTH_LONG).show();
        Intent intent = new Intent(getApplicationContext(), MainActivity.class);
        finish();
        startActivity(intent);
    }

}

File: app/src/main/java/com/goel/attendancetracker/activities/NewOrganisationActivity.kt
package com.goel.attendancetracker.activities

import android.content.ContentValues
import android.content.Intent
import android.database.sqlite.SQLiteException
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.goel.attendancetracker.databinding.ActivityNewOrganisationBinding
import com.goel.attendancetracker.models.OrganisationsModel
import com.goel.attendancetracker.utils.Constants
import com.goel.attendancetracker.utils.database.DatabaseHandler
import java.lang.NullPointerException

class NewOrganisationActivity : AppCompatActivity() {

    private lateinit var binding: ActivityNewOrganisationBinding
    private lateinit var databaseHandler: DatabaseHandler

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityNewOrganisationBinding.inflate(layoutInflater)
        setContentView(binding.root)
        databaseHandler = DatabaseHandler(this@NewOrganisationActivity)
        binding.newOrganisationSubmit.setOnClickListener { submitNewOrganisation() }
    }

    private fun submitNewOrganisation() {
        val nameValidity = Constants.getNameValidity(binding.editNewOrganisationName.text.toString())
        val targetValidity = Constants.getTargetValidity(binding.editNewOrganisationTarget.text.toString())

        if (nameValidity.containsKey(false)) {
            binding.editNewOrganisationName.error = nameValidity[false]
            return
        }
        if (targetValidity.containsKey(false)) {
            binding.editNewOrganisationTarget.error = targetValidity[false]
            return
        }
        val organisationName: String
        val organisationTarget = try {
            organisationName = nameValidity[true]!!
            targetValidity[true]?.toInt()!!
        } catch (e: NullPointerException) {
            return
        }
        val newOrganisation = OrganisationsModel(name = organisationName, target = organisationTarget)
        try {
            databaseHandler.createNewOrganisation(newOrganisation)
            insertOrganisation(newOrganisation)
        } catch (e: SQLiteException) {
            binding.editNewOrganisationName.error = "Organisation already exists"
            Toast.makeText(this, "Organisation already exists", Toast.LENGTH_SHORT).show()
        }
    }

    private fun insertOrganisation(newOrganisation: OrganisationsModel) {
        val values = ContentValues()
        values.put(Constants.NAME, newOrganisation.name)
        values.put(Constants.TARGET, newOrganisation.target)
        values.put(Constants.ATTENDANCE, newOrganisation.attendance)
        databaseHandler.insertOrganisation(values)
        Toast.makeText(this, "Added " + newOrganisation.name + " Successfully", Toast.LENGTH_LONG)
            .show()
        val intent = Intent(applicationContext, MainActivity::class.java)
        finish()
        startActivity(intent)
    }
}